webapp/channels/src/components/admin_console/file_upload_setting.tsx

```diff
@@ -19,134 +19,143 @@ type Props = {
     error?: string;
 }
 
-type State = {
-    fileName: null|string;
-    fileSelected: boolean;
-    serverError?: string;
-    uploading: boolean;
-}
+const FileUploadSetting = ({
+    id,
+    error: errorFromProps,
+    label,
+    helpText,
+    disabled,
+    fileType,
+    uploadingText,
+    onSubmit,
+}: Props) => {
+    const [fileNameFromState, setFileNameFromState] = useState<string | null>(null);
+
+    const [isUploading, setIsUploading] = useState(false);
+    const [isFileSelected, setIsFileSelected] = useState(false);
 
-export default class FileUploadSetting extends React.PureComponent<Props, State> {
-    fileInputRef = React.createRef<HTMLInputElement>();
+    const fileInputRef = useRef<HTMLInputElement>(null);
 
     // Helps prevent setting state after component is unmounted, for usage when this component is wrapped by a custom setting
-    isMounted = false;
+    const isMounted = useRef(false);
 
-    constructor(props: Props) {
-        super(props);
+    useEffect(() => {
+        isMounted.current = true;
 
-        this.state = {
-            fileName: null,
-            serverError: props.error,
```

Comment:
`serverError` state in the class component is never updated and is set to the `error` received in the props.

Since the state is never updated and is only used when checking if it has a value (in the `render` function, line 84), it might as well be the same as using the prop directly, so that's what I went with.